test/e2e/case1_audit_from_cache_test.go

```diff
@@ -0,0 +1,370 @@
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/open-policy-agent/gatekeeper/v3/apis/config/v1alpha1"
+	"golang.org/x/exp/slices"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	case1GatekeeperYaml             string = "../resources/case1_audit_from_cache/gatekeeper.yaml"
+	case1TemplateYaml               string = "../resources/case1_audit_from_cache/template.yaml"
+	case1ConstraintPodYaml          string = "../resources/case1_audit_from_cache/constraint-pod.yaml"
+	case1ConstraintPod2Yaml         string = "../resources/case1_audit_from_cache/constraint-pod-2.yaml"
+	case1ConstraintIngressYaml      string = "../resources/case1_audit_from_cache/constraint-ingress.yaml"
+	case1ConstraintStorageclassYaml string = "../resources/case1_audit_from_cache/constraint-storageclass.yaml"
+	case1PodYaml                    string = "../resources/case1_audit_from_cache/pod.yaml"
+	allowNamespace                  string = "case1-allow"
+	denyNamespace                   string = "case1-deny"
+	case1ConstraintUpdateYaml       string = "../resources/case1_audit_from_cache/constraint-update.yaml"
+	case1ConstraintUpdateChangeYaml string = "../resources/case1_audit_from_cache/constraint-update-change.yaml"
+	case1ConstraintWrongYaml        string = "../resources/case1_audit_from_cache/constraint-wrong.yaml"
+)
+
+var constraintGVR = schema.GroupVersionResource{
+	Group:    "constraints.gatekeeper.sh",
+	Version:  "v1beta1",
+	Resource: "case1template",
+}
+
+var templateGVR = schema.GroupVersionResource{
+	Group:    "templates.gatekeeper.sh",
+	Version:  "v1",
+	Resource: "constrainttemplates",
+}
+
+var _ = Describe("Test auditFromCache", Ordered, func() {
+	BeforeAll(func() {
+		if !useExistingCluster() {
+			Skip("Test requires existing cluster. Set environment variable USE_EXISTING_CLUSTER=true and try again.")
+		}
+
+		By("Create namespaces to compare")
+		Kubectl("create", "ns", allowNamespace)
+		Kubectl("create", "ns", denyNamespace)
+
+		By("Create a gatekeeper resource")
+		_, err := KubectlWithOutput("apply", "-f", case1GatekeeperYaml)
+		Expect(err).ShouldNot(HaveOccurred())
+		// Need enough time until gatekeeper is up
+		ctlDeployment := GetWithTimeout(clientHubDynamic, deploymentGVR,
+			"gatekeeper-controller-manager", gatekeeperNamespace, true, 150)
+		Expect(ctlDeployment).NotTo(BeNil())
+
+		Eventually(func(g Gomega) {
+			auditDeployment := GetWithTimeout(clientHubDynamic, deploymentGVR,
+				"gatekeeper-audit", gatekeeperNamespace, true, 60)
+			g.Expect(auditDeployment).NotTo(BeNil())
+
+			availableReplicas, _, err := unstructured.NestedInt64(auditDeployment.Object, "status", "availableReplicas")
+			g.Expect(err).ShouldNot(HaveOccurred())
+			g.Expect(availableReplicas).Should(BeNumerically(">", 0))
+		}, 2*time.Minute, 10*time.Second).Should(Succeed())
```

Comment:
I think checking every 10 seconds is too long.
```suggestion
		}, 2*time.Minute, 2 * time.Second).Should(Succeed())
```